test/Node/NpmLockSpec.hs

```diff
@@ -170,3 +193,24 @@ spec = do
         , (packageFive, packageSix)
         ]
         graph
+  describe "parsing package-json.lock" $ do
+    currentDir <- runIO getCurrentDir
+    let testDir = currentDir </> $(mkRelDir "test/Node/testdata")
+
+    it' "Should ignore \"resolved\": <bool> in package-lock.json" $ do
+      let packageLock = testDir </> $(mkRelFile "boolean-resolved-package-lock.json")
+      NpmPackageJson{packageDependencies = packageDependencies} <- readContentsJson packageLock
+      let foo = Map.lookup "foo" packageDependencies >>= unNpmResolved . depResolved
```

Comment:
[nit]: This is one of the cases where flipped operators make things more readable.  Currently, the code has to be read as:

```haskell
firstOp >>= thirdOp . secondOp
```
However, we can use the right-to-left style of the `.` operator and justify using `=<<` (which is normally discouraged in the style guide), because then the ordering flows neatly from one side of the expression to the other.
```suggestion
      let foo = unNpmResolved . depResolved =<< Map.lookup "foo" packageDependencies
```

If you choose to add this, probably best to change the other tests as well.

Reply:
Thanks, I have incorporated this suggestion.

Reply:
I've added this to the style guide in #776